Source/WebCore/rendering/BackgroundPainter.cpp

```diff
@@ -512,9 +512,11 @@ BackgroundImageGeometry BackgroundPainter::calculateBackgroundImageGeometry(cons
     LayoutSize positioningAreaSize;
     // Determine the background positioning area and set destination rect to the background painting area.
     // Destination rect will be adjusted later if the background is non-repeating.
-    // FIXME: transforms spec says that fixed backgrounds behave like scroll inside transforms. https://bugs.webkit.org/show_bug.cgi?id=15679
+    auto enclosing = renderer.enclosingLayer();
+    bool isTransformed = renderer.hasTransform() || (enclosing ? enclosing->hasTransformedAncestor() : false);
+    bool fixedAttachment = fillLayer.attachment() == FillAttachment::FixedBackground && !isTransformed && fillLayer.image();
```

Comment:
(just a question): why are we now checking for `fillLayer.image()` ?

Reply:
Actually I don't think we do, removing it.